### DISCUSSION TITLE
[Seção 3.14] - Interpolação, locals e outputs - Error creating S3 bucket: InvalidBucketName: The specified bucket is not valid.

### DISCUSSION THREAD

**Discussion by ironftw** [chgasparoto/curso-aws-com-terraform#103] [Q&A] [answered]
Estou recebendo erro na hora de criar o bucket.

Segue mensagem de erro durante o terraform apply e o conteúdo do vars.

![erro_s3](https://user-images.githubusercontent.com/19733960/157095247-2dbb062f-7ed6-498e-b76b-33c41cbe6fb7.jpg)
![vars](https://user-images.githubusercontent.com/19733960/157095265-aea04849-f669-47e0-b3b4-87822cfd8691.jpg)

adding random.tf and main.tf
![random tf](https://user-images.githubusercontent.com/19733960/157099800-046856e2-99b4-4270-bd33-e1e34fdf9c95.jpg)
![main tf](https://user-images.githubusercontent.com/19733960/157100119-d45d5994-80fb-47ff-812f-6c1731c99126.jpg)

**Comment by developando** [comment, marked as answer]
Boa noite,

Estava com o mesmo problema e consegui localizar o erro, de acordo com a politica de nomes de buckets da aws, ele apenas 
aceita letras minusculas:

**https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html**

Trecho
**Os nomes dos buckets podem consistir apenas em letras minúsculas, números, pontos (.) e hífens (-).**


Por algum motivo eu coloquei"Dev" com letra maiuscula e assim era retornado o erro, após ajustar o "D" para "d" o bucket é criado sem problemas: 


```
variable "environment" {
  type        = string
  description = ""
  default     = "dev"

}
```

**Reply by chgasparoto** [reply]
@developando isso mesmo. Muito obrigado pela resposta!

**Reply by ironftw** [reply]
O problema era esse! Detalhezinho sacana.

Obrigado pessoal!
![buckets3](https://user-images.githubusercontent.com/19733960/157234122-31a5ae52-313e-4117-8582-8f59307e89fe.jpg)
